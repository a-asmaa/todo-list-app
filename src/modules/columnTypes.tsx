export interface ColumnsTypes {

    [key: string]: {
        title: string,
        items: Item[]
    }
}

export interface Item {
    id: string,
    name: string,
    description: string,
    isEdit: boolean
}