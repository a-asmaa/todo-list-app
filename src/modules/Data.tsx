import { ColumnsTypes } from "./columnTypes";

export const columnsTypes: ColumnsTypes = {
    todo: {
        title: "📝 Todo",
        items: []
    },
    progress: {
        title: "⌛️ In Progress",
        items: []
    },
    done: {
        title: "🏆 Done",
        items: []
    }
};