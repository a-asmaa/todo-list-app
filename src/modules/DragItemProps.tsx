import { Item } from "./columnTypes";

export interface IDraggableItems {
    index: number,
    keyValue: string,
    handleEdit: any,
    handleInLineEdit: any,
    item: Item,
    handleDelete: any
}