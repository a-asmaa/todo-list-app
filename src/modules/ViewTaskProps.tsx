import { Item } from "./columnTypes";

export interface IViewTask {
    show: boolean,
    onHide: any,
    onSave: any,
    item: Item | null,
    type: string
}