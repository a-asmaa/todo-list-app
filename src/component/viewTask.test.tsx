import { getByTestId, render, screen } from "@testing-library/react"
import { IViewTask } from "../modules/ViewTaskProps"
import ViewTask from "./viewTask"


it('Header Component should match with snapshot', () => {

    const props: IViewTask = {
        handleSave: () => console.log("saveFn"),
        item: null,
        onHide: () => console.log('closeModel'),
        type: "todo",
        show: true
    }
    expect(render(<ViewTask {...props} />)).toMatchSnapshot() // snapshot from component 
})

it('Test Modal rendering', () => {
    // screen.getByText('Track your Tasks')
    // console.log(screen.queryByTestId('button'));

    // expect(screen.queryByTestId('button')).toBeVisible();
})   
