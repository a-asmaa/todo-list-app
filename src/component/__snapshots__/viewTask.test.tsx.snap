// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-labelledby="contained-modal-title-vcenter"
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-lg modal-dialog-centered"
        type="todo"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-body"
          >
            <div
              class="mb-4 justify-content-between hstack gap-3"
            >
              <h5>
                 Task Details 
              </h5>
              <i
                class="bi bi-pencil-square"
                role="button"
              />
            </div>
            <h6
              class="form-label"
            >
              ðŸ–Š Task Name 
            </h6>
            <p
              class="mx-4 mb-3"
            >
               
                
            </p>
            <h6
              class="form-label"
            >
               ðŸ“ƒ Description 
            </h6>
            <p
              class="mx-4"
            >
               
               
            </p>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-primary"
              hidden=""
              type="button"
            >
              Save
            </button>
            <button
              class="btn btn-primary"
              data-testid="button"
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
